### DISCUSSION TITLE
Create texture from canvas

### DISCUSSION THREAD

**Discussion by lucadalli** [google/model-viewer#3211] [Ideas]
At the moment, creating a texture from canvas is possible by passing the data URL or an object URL to the `createTexture` method. For larger textures this adds quite a bit of overhead (~100ms).
THREE natively supports creating a texture from canvas via [CanvasTexture](https://threejs.org/docs/#api/en/textures/CanvasTexture). Is this something we can leverage to allow `createTexture` to accept canvas input?

**Comment by samaneh-kazemi** [comment]
Hi, looking at how CanvasTexture is implemented and the documentation, it should have similar overhead since it calls Texture constructor. This is also stated in their doc you shared. The 'createTexture' is using 'Texture' so I'm not sure if even if we support CanvasTexture it helps with the overhead.

[CanvasTexture code ](https://github.com/mrdoob/three.js/blob/master/src/textures/CanvasTexture.js)
```
		super( canvas, mapping, wrapS, wrapT, magFilter, minFilter, format, type, anisotropy );

		this.needsUpdate = true;
```

**Reply by elalish** [reply]
I believe @lucadalli is making that point that Three's `Texture` object can take a canvas element and handle it more efficiently than a URL. It's possible we are artificially restricting ourselves. @lucadalli would you be interesting in trying a PR for this?

**Reply by lucadalli** [reply]
@elalish I don't mind giving it a go.
